Properly handle node version and min compatibility versions

dbwiddis opened the pull request:
### Description

OpenSearch nodes return the current node version in the handshake response.

We incorrectly returned the minimum compatibility version of the sending node, which happened to work on 3.0.0 but doesn't work on 2.x.

### Issues Resolved

Fixes #166

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

dblock reviewed (commented):
Add a test?

dbwiddis commented:
> Add a test?

My brief attempt to wake up my Python build environment still has bugs and I can't compile/run locally.  I have zero bandwidth for this until the long weekend.   I'll get to it then, your call on whether to merge without a failing test on the old code that I'm not even sure where it will go yet.

dblock commented:
Thanks for your help @dbwiddis. 

@mzainab24 you can take this change in the meantime to get unblocked?

mzainab24 commented:
Thanks I will test it in my environment. And post the results here.

dbwiddis commented:
Updated and added tests.  We probably could have kept most of the code that returned the same min compatibility version as the sending node and only changed the actual version return in the initial TCP handshake... but this is now a bit more consistent with an OpenSearch (extension or other) node
